### DISCUSSION TITLE
housekeeping: Error:1220

### DISCUSSION THREAD

**Discussion by DennisGlobal** [ipxe/ipxe#823] [Q&A]
My issue,
After clarification: Dependencies issue
https://github.com/ipxe/ipxe/issues/821

This is still my 1st compile,
I now got it running up untill i run in an EFI/UEFI error?

make bin/{undionly,ipxe}.{,k,kk}pxe EMBED=ipxescript
  [VERSION] bin/version.undionly.pxe.o
  [LD] bin/undionly.pxe.tmp
ld: bin/blib.a(efi_image.o): in function `efi_image_probe':
/root/ipxe/src/image/efi_image.c:312: undefined reference to `efi_systab'
ld: bin/blib.a(efi_image.o): in function `efi_image_exec':
/root/ipxe/src/image/efi_image.c:136: undefined reference to `efi_systab'
ld: bin/blib.a(efi_init.o): in function `efi_raise_tpl':
/root/ipxe/src/interface/efi/efi_init.c:395: undefined reference to `efi_systab'
ld: bin/blib.a(efi_init.o): in function `efi_restore_tpl':
/root/ipxe/src/interface/efi/efi_init.c:411: undefined reference to `efi_systab'
ld: bin/blib.a(efi_pxe.o): in function `efi_pxe_install':
/root/ipxe/src/interface/efi/efi_pxe.c:1591: undefined reference to `efi_systab'
ld: bin/blib.a(efi_pxe.o):/root/ipxe/src/interface/efi/efi_pxe.c:1675: more undefined references to `efi_systab' follow
ld: bin/undionly.pxe.tmp: hidden symbol `efi_systab' isn't defined
ld: final link failed: bad value
make: *** [Makefile.housekeeping:1220: bin/undionly.pxe.tmp] Error 1
rm bin/version.undionly.pxe.o
============================================
As i think it works, i just need to "#undef" or "//#undef" all EFI-like values in the general.h?

**Comment by NiKiZe** [comment]
You should not modify configs unless you HAVE to.
Start with clean config, change one thing at a time.

The only reasonable target for undionly is .kpxe
Undionly.pxe will not work, and kkpxe should not be used.

**Reply by DennisGlobal** [reply]
okay, i tried to compile from the Clean config (just redownloaded the github)

followed the 2-liner on the website:
cd ipxe/src
make

but even then i get an error

util/genfsimg: could not find isolinux.bin
make: *** [arch/x86/Makefile.pcbios:61: bin/ipxe.iso] Error 1
rm bin/ipxe.dsk.zinfo bin/ipxe.lkrn.bin bin/ipxe.lkrn.zinfo bin/ipxe.lkrn.zbin bin/ipxe.dsk.zbin bin/version.ipxe.dsk.o bin/ipxe.dsk.bin bin/version.ipxe.lkrn.o

Isnt there an step-by-step guide on how to get the ipxe files working?

the "Undionly.pxe will not work, and kkpxe should not be used." part is coming from the following Forum:
https://wiki.fogproject.org/wiki/index.php/Building_undionly.kpxe

I can look for an 'isolinux.bin' file, and drop it into the folder and try again.
but im not sure if it works that way.

edit: found one,
not working while i put it into:
/arch/x86/
or
/bin/

**Reply by NiKiZe** [reply]
So you are building ipxe.iso, do you need that? If so install the dependencies as described on the download page.

You are blindly following some random post on different forum instead of listening to the facts of the iPXE project?

You are assuming that you are missing files from iPXE repo, you are not, however you are missing dependencies that are required to build. You need to install those dependencies on your machine. It is up to you to find out which packages it is on your specific system.

**Reply by DennisGlobal** [reply]
> So you are building ipxe.iso, do you need that? If so install the dependencies as described on the download page.
> 
> You are blindly following some random post on different forum instead of listening to the facts of the iPXE project?
> 
> You are assuming that you are missing files from iPXE repo, you are not, however you are missing dependencies that are required to build. You need to install those dependencies on your machine. It is up to you to find out which packages it is on your specific system.

yes im trying to get an iPXE bootfiles build,
But i did get those Dependencies in place.

I got that corrected on this list:
https://github.com/ipxe/ipxe/issues/821#issuecomment-1333435537

there was an dependecy on the website called "liblzma or xz header files"
those werent found on that name, but i dused liblzma-dev, and the error was solved, bút raised another one instead :(

So im now Blindly following this random made post, and still got errors.
even at a clean downloaded, installed state.

**Reply by NiKiZe** [reply]
So you mean that iPXE should match its dependency requirement on what it is called in your distro? What about all other distros then?

You are building iPXE.iso, it requires tools to generate the iso, you need to install those tools. This is NOT MISSING from iPXE. You need to install the required packages.

Or, stop building ipxe.iso ;)

**Reply by DennisGlobal** [reply]
> So you mean that iPXE should match its dependency requirement on what it is called in your distro? What about all other distros then?
> 
> You are building iPXE.iso, it requires tools to generate the iso, you need to install those tools. This is NOT MISSING from iPXE. You need to install the required packages.
> 
> Or, stop building ipxe.iso ;)

hahahaha
what a joke this comment is

please read it again

as i mentioned:
I have installed ALL dependency requirements on my distro, being: DEBIAN 11.
 thus leading ME to this Error: > util/genfsimg: could not find isolinux.bin
make: *** [arch/x86/Makefile.pcbios:61: bin/ipxe.iso] Error 1
 
there was an dependecy on the website called "liblzma or xz header files"
those werent found on that name, but i dused liblzma-dev, and the error was solved, bút raised another one instead :(

Mainly being: >> util/genfsimg: could not find isolinux.bin

But its easy to say from an 20y old software age view "Or, stop building ipxe.iso ;)"
tháts a good advice...

**Reply by stappersg** [reply]
_How To Ask Questions The Smart Way_ has a section [How To Answer Questions in a Helpful Way](http://www.catb.org/~esr/faqs/smart-questions.html#idm667). But yes, `NiKiZe` does bully, sad but true. My advice is to focus on getting iPXE building on your host, _don't fight back, fight forward_.

**Reply by NiKiZe** [reply]
> > So you mean that iPXE should match its dependency requirement on what it is called in your distro? What about all other distros then?
> > You are building iPXE.iso, it requires tools to generate the iso, you need to install those tools. This is NOT MISSING from iPXE. You need to install the required packages.
> > Or, stop building ipxe.iso ;)
> 
> hahahaha what a joke this comment is
> 
> please read it again
> 
> as i mentioned: I have installed ALL dependency requirements on my distro, being: DEBIAN 11. thus leading ME to this Error: > util/genfsimg: could not find isolinux.bin make: *** [arch/x86/Makefile.pcbios:61: bin/ipxe.iso] Error 1
> 
> there was an dependecy on the website called "liblzma or xz header files" those werent found on that name, but i dused liblzma-dev, and the error was solved, bút raised another one instead :(
> 
> Mainly being: >> util/genfsimg: could not find isolinux.bin
> 
> But its easy to say from an 20y old software age view "Or, stop building ipxe.iso ;)" tháts a good advice...

I was asking if you needed iPXE.iso? Will you be burning a CD? Often there is better ways to boot, such as ipxe.usb.

There is multiple dependencies, depending on your build target, for iso you need isolinux.bin, which often, but not always, is part of syslinux.

**Comment by stappersg** [comment]
> Isn't there an step-by-step guide on how to get the ipxe files working?

![afbeelding](https://user-images.githubusercontent.com/35434094/205091175-8864cd0d-c1c7-4712-b140-b4138ceed349.png)

> following Forum: https://wiki.fogproject.org/wiki/index.php/Building_undionly.kpxe

So contact them.  ( Bark at the right tree ( Stop barking at this tree ))


>  ...`isolinux.bin` ... not sure ... not working

Also from https://ipxe.org/download#source_code

_You can build an iPXE bootable USB key image using:_

```
make bin/ipxe.usb
```

And report back.

That _reporting back_ is important to us. It tells us (people that have iPXE working) that you have a build success.  Yeah, not a _iPXE ISO_, but you do have an _iPXE_.  That _iPXE binary_ is a milestone for your further _netbooting journey_.

**Reply by DennisGlobal** [reply]
> > Isn't there an step-by-step guide on how to get the ipxe files working?
> 
> ![afbeelding](https://user-images.githubusercontent.com/35434094/205091175-8864cd0d-c1c7-4712-b140-b4138ceed349.png)
> 
> > following Forum: https://wiki.fogproject.org/wiki/index.php/Building_undionly.kpxe
> 
> So contact them. ( Bark at the right tree ( Stop barking at this tree ))

i didnt bark up the wrong tree, but came for advice.
advice here was "You should not modify configs unless you HAVE to."
So like a good dog as u see me, i dropped the fog wiki-steps.

> > ...`isolinux.bin` ... not sure ... not working

It seems to be missing, and i use the clean pull from Gihub, and have the Dependencies installed.
Also, just copyd an isolinux file uit of an bootimage, placed it into 2 different folder i though that were missing it.
but it didnt work.

> Also from https://ipxe.org/download#source_code
> 
> _You can build an iPXE bootable USB key image using:_
> 
> ```
> make bin/ipxe.usb
> ```
> 
> And report back.

this is the 'guide' i meant.
I will try this and hope that it works.

> That _reporting back_ is important to us. It tells us (people that have iPXE working) that you have a build success. Yeah, not a _iPXE ISO_, but you do have an _iPXE_. That _iPXE binary_ is a milestone for your further _netbooting journey_.

That is why im 'barking' up this tree.
iPXE looks and feel great, but the Guide's? Community is thin and hard to find. (atleast for me)

**Reply by stappersg** [reply]
> > So im now Blindly following this random made post, and still got errors.

No worries.  Remember how you learnt walking, many downs and ups.


> > even at a clean downloaded, installed state.

Yes, build dependencies are not available on a clean system.


> You need to install the required packages.

Tell us about how you are doing that.  Share with us which commands you are using.

> > > those werent found on that name, but I used liblzma-dev, and the error was solved,

That indicates progress. Please include with which commands the progress was made.

> > > bút raised another one instead :(

Report **which** error.



> That is why im 'barking' up this tree.

No worries, the discussion continues.


> iPXE looks and feel great,

Yes, iPXE is great.  That is why I make the effort to contribute. The why I reply here.


> but the Guide's?
> Community is thin and hard to find. (atleast for me)

That feels like unaware of [how to ask smart questions](http://www.catb.org/~esr/faqs/smart-questions.html).  Er is ook een [vertaling naar het Nederlands](http://docs.jaspervries.nl/smart-questions.html).


Groeten
Geert Stappers
Looking forward to further feedback.
Please, pretty please, include the _"what was done"_ to the _"now I got something I don't understand yet"_.
